llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -25,6 +26,11 @@ using namespace PatternMatch;
 
 #define DEBUG_TYPE "instcombine"
 
+// Controls the maximum memory allocation to 32-bit or 4GB.
+static cl::opt<bool> PointerAllocationsLimitedto32bit(
+    "alloc-limit-32bit", cl::init(false),
+    cl::desc("A Pointer with max 32bit allocs"));
```

Comment:
I'm not willing to add this flag.

Reply:
I agree.
WARNING: Adding this flag may violate SPEC's rule.
See also https://www.spec.org/cpu2017/Docs/runrules.html#rule_1.4 and https://www.spec.org/cpu2017/results/res2023q1/cpu2017-20221223-33220.html.

Reply:
I have provided my understanding of this code pattern and dont quite agree that it violates a benchmark rule or criteria.